### ISSUE TITLE
how to set the  anticlastic curvature optics

### ISSUE THREAD

**Issue by tyjisinap** [srio/shadow3#23]
Dear Srio,
I want to raytrace the high energy laue monochromator, but how to set the anticlastic curvature optics? I think I can use a external file at the modified surface option.  and I also found that there is pole location parameter for the torodal mirror , is that for a anticlastic curvature? and if so,how to use it?  I try to set it but i don't know whether it is right or not.  Thank you very much!

Best regards!
Yajun Tong
SSRF, China

**Comment by tyjisinap** [comment]
Dear Srio,
I have got the answer from the shadow help file, Thank you very much！

**Comment by lucarebuffi** [comment]
Pity, I would like to know it...

**Comment by tyjisinap** [comment]
Dear Luca,
This picture from help file shows the defination of the torodal mirror. from this pic, the upper/Inner and Lower/inner cases give anticlastic curvature and pay attention that the actual radius was different from the input value.
Best regards.
 
2016-11-08


tongyajun
 
From: Luca Rebuffi
Date: 2016-11-07 18:24
To: srio/shadow3
CC: Yajun Tong; State change
Subject: Re: [srio/shadow3] how to set the anticlastic curvature optics (#23)
Pity, I would like to know it...

**Comment by tyjisinap** [comment]
there is still another question,  I set the torodal crystal, when the option is upper /inner, the rays go in a right direction, because when I plot the star.01 the rays are around the zero, and also can focus(I filtered the lower reflectivity rays to be lost flag in the shadowvui, and not need in the shadowoui , and can see the diffracting rays focusing). but when i change to lower/inner, the rays seems go in wrong direction, because the rays are not around the zero. and the crystal is set to auto setting. I think this change only can change the focusing property, and not change the main axis direction. why change?  is it a bug? or I need to change other setting?

**Comment by srio** [comment]
Just think that these systems (except the concave/concave) are defocusing so the effects of source size, divergence etc will be important. 

I tried a point source with a very small divergence (e.g 1 urad) and a simple toroid (in autofocusing p=100 cm q=200 cm) and the image is always centered.

**Comment by tyjisinap** [comment]
yes, if it is as a mirror, everything is ok, I have tried all the options. but when I set it as a asymmetric cutting laue crystal and concave/convex, it is not centered. I changed the reflect angle to make it center, and I found that the rays go to a strange direction. I don't know whether it can trace such a crystal?  I found that for a convex cylinder crystal, it can focus the diffraction rays. Thank you very much!
2016-11-09


tongyajun
 
From: Manuel Sanchez del Rio
Date: 2016-11-09 15:15
To: srio/shadow3
CC: Yajun Tong; State change
Subject: Re: [srio/shadow3] how to set the anticlastic curvature optics (#23)
Just think that these systems (except the concave/concave) are defocusing so the effects of source size, divergence etc will be important. 
I tried a point source with a very small divergence (e.g 1 urad) and a simple toroid (in autofocusing p=100 cm q=200 cm) and the image is always centered.
